prog/aws/instance.rb

```diff
@@ -181,7 +181,15 @@ def before_run
       raise
     rescue Aws::EC2::Errors::InsufficientInstanceCapacity => e
       if is_runner? && (runner = GithubRunner[vm_id: vm.id])
-        Clog.emit("insufficient instance capacity") { {insufficient_instance_capacity: {vm:, message: e.message}} }
+        families = frame["alternative_families"]
+        next_family = if families && !families.empty? && (index = families.index(vm.family) || -1) < families.length - 1
+          families[index + 1]
+        end
+        Clog.emit("insufficient instance capacity") { {insufficient_instance_capacity: {vm:, next_family:, message: e.message}} }
+        if next_family
+          vm.update(family: next_family)
+          nap 1
```

Comment:
Does the Amazon API react negatively to 4 API requests in quick succession? If not, maybe `nap 0` would be better here.